grpc/src/types/auth.rs

```diff
@@ -0,0 +1,92 @@
+use pbjson_types::Any;
+use prost::{Message, Name};
+
+use celestia_proto::cosmos::auth::v1beta1::{
+    QueryAccountRequest, QueryAccountResponse, QueryAccountsRequest, QueryAccountsResponse,
+    QueryParamsRequest as QueryAuthParamsRequest, QueryParamsResponse as QueryAuthParamsResponse,
+};
+use celestia_types::auth::{
+    AuthParams, BaseAccount, ModuleAccount, RawBaseAccount, RawModuleAccount,
+};
+use celestia_types::state::Address;
+
+use crate::types::make_empty_params;
+use crate::types::{FromGrpcResponse, IntoGrpcParam};
+use crate::Error;
+
+/// Enum representing different types of account
+#[derive(Debug, PartialEq)]
+#[non_exhaustive]
```

Comment:
My thinking was to be future compatible in case new account type pops up. Upon retrospection I'm not sure that's needed and we can just release breaking version in that case.